src/lib/GenericPredictor.cc

```diff
@@ -4,17 +4,24 @@
 
 namespace simeng {
 
-GenericPredictor::GenericPredictor(YAML::Node config)
-    : btbBits_(config["Branch-Predictor"]["BTB-Tag-Bits"].as<uint64_t>()),
-      btb_(1 << btbBits_,
-           {config["Branch-Predictor"]["Fallback-Static-Predictor"]
-                .as<uint16_t>(),
-            0}),
-      satCntBits_(
-          config["Branch-Predictor"]["Saturating-Count-Bits"].as<uint64_t>()),
-      globalHistoryLength_(
-          config["Branch-Predictor"]["Global-History-Length"].as<uint64_t>()),
-      rasSize_(config["Branch-Predictor"]["RAS-entries"].as<uint64_t>()) {
+GenericPredictor::GenericPredictor(ryml::ConstNodeRef config) {
+  btbBits_ = config["Branch-Predictor"]["BTB-Tag-Bits"].as<uint8_t>();
+  // Create branch prediction structures
+  satCntBits_ =
+      config["Branch-Predictor"]["Saturating-Count-Bits"].as<uint8_t>();
+  globalHistoryLength_ =
+      config["Branch-Predictor"]["Global-History-Length"].as<uint16_t>();
+  rasSize_ = config["Branch-Predictor"]["RAS-entries"].as<uint16_t>();
```

Comment:
These 4 variables could be added to initialiser list

Reply:
Good spot. Also changed in `SpecialFilDenGen`